### ISSUE TITLE
bug: S3 DeleteObjectTagging returns 500 - Internal Error 

### ISSUE THREAD

**Issue by anna-mazhar** [localstack/localstack#10330]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

- When attempting to delete object tagging for a non-existent key using the delete_object_tagging boto SDK method, LocalStack fails with a 500 status code. 
- The error code is "InternalError" with the message: "exception while calling s3.DeleteObjectTagging: 'NoneType' object has no attribute 'arn'".

### Expected Behavior

- The expected behavior, as observed in the cloud, is to fail with a 404 status code indicating a "NoSuchKey" error. The error message should state "The specified key does not exist." 
- The above cloud response was captured by executing the same set of APIs on the cloud.

### How are you starting LocalStack?

With a `docker run` command

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker run localstack/localstack

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

    s3_client = boto3.client('s3', endpoint_url='http://localhost:4566')
    response = s3_client.delete_object_tagging(
        Bucket='test_bucket',
        Key='nonexistent-key'
    )
The query is made for a non existent key.

### Environment

```markdown
- OS: Ubuntu 22.04.2 LTS
- LocalStack: 2.1.0
- boto3: 1.28.6
```

**Comment by bentsku** [comment]
Hello @anna-mazhar and thanks for your report! Like #10331, it seems this issue has been fixed in the meantime in versions > 3.

I could not reproduce:

```python
response = s3_client.delete_object_tagging(
    Bucket=bucket,
    Key='nonexistent-key'
)
> botocore.errorfactory.NoSuchKey: An error occurred (NoSuchKey) when calling the DeleteObjectTagging operation (reached max retries: 0): The specified key does not exist.
```

Could you please confirm that it now works? Thanks!

**Comment by anna-mazhar** [comment]
Hi @bentsku, thanks for the swift response! I confirm that this has been fixed in 3.2.0.
